### PR TITLE
Undercity object despawns

### LINKED ISSUE #920: Enchanting suppliers/trade suppliers should sell Mystical Powder
**Describe the bug**
Mystical Powder (6216) is required to craft enchants. However, it is currently not sold by any vendor and there is no other way for the player to obtain it. This renders the enchanting profession unusable.

**Steps to reproduce**
1. Learn enchanting
2. Check any tradeskill or enchanting related vendor, anywhere in the world.
3. Be very sad.

**Expected behavior**
Mystical Powder should be readily available from any enchanting or profession related vendor just like any other common trade skill resource. Given than you can learn enchanting very early in the starting areas of each race one can assume *every* related vendor indeed sold this item.

**Sources**
According to [this source](https://crawler.thealphaproject.eu/mnt/crawler/media/Text/Various/alpha-tester-report-march-18-2004.txt), the undead push - which is 0.5.3 - included the enchanting profession ready to use:

> This was added to the game in the undead push, and **that's when I tried it**. Magical effects can be added to various pieces of armor – the effect and the target depend on the recipe – using herbs, magical dust and copper and silver rods. In order to keep things balanced, permanent enchantments use a lot of herbs, and most of the skilling up recipes are temporary imbuing recipes.

It's impossible to "try it" if a core ingredient is missing from the game.

### LINKED ISSUE #921: TODO: Despawn objects in Apothecary Quarter, Undercity
**Describe what needs to be implemented or enhanced**
In the Apothecary Quarter of Undercity you can find this nice arrangement of Shane Cubes:

image

This is bleed-in from vanilla where these objects were related to the quest "Testing for Corruption: Felwood" which obviously wasn't in 0.5.3. [This image](https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Cities/Undercity/worldofwarcraft_042604_023.jpg) from beta (or alpha?) shows the equipment wasn't present back then.

**Expected end result**
The objects should be gone and the Apothecary Quarter clean again!

**Guidance**
Despawn objects:
- entry 175265 and 174848 (the testing equipment)
- entry 58610 (the hanging alchemist)

### PR DESCRIPTION
**Pull request by mindphluxnet**
Closes #921 

- Despawns three objects - 2x testing equipment, 1 x hanging alchemist - from Undercity as they weren't present in 0.5.3 (they were added with Felwood)

### CODE REVIEW

**Comment by devw4r** [comment]
Remember to add the `insert into applied_updates` part. 👍

**Comment by mindphluxnet** [comment]
Yep, sorry, copy&paste got me.

**Comment by GrenderG** [comment]
I added some notes here @mindphluxnet https://github.com/The-Alpha-Project/alpha-core/issues/920#issuecomment-1456399294 Do you want to include them in this PR? Or you want me to merge it and then they can later be included?

Basically inventories of Trade Suppliers need to be changed to match the sources we have, `<Enchanting Supplies>` need to be changed to `<Enchanter>`, and maybe rework their inventories too to remove the `Bundle of Wood`.

**Comment by mindphluxnet** [comment]
I included the tradeskill vendors now and will do the subname and other inventory changes later.

**Comment by GrenderG** [comment]
> I included the tradeskill vendors now and will do the subname and other inventory changes later.

Sounds good, thanks!

**Comment by mindphluxnet** [comment]
I'm getting merge conflicts, no idea what's wrong here.

**Comment by GrenderG** [comment]
> I'm getting merge conflicts, no idea what's wrong here.

It's because I merged https://github.com/The-Alpha-Project/alpha-core/pull/937. You need to pull the file and include its changes here.

**Comment by GrenderG** [comment]
Anyway if you want we can leave this PR for the despawning of gobjects and you can open a new one for inventories later.

**Comment by mindphluxnet** [comment]
Done, should work now.

**Comment by GrenderG** [comment]
I'd redo from scratch the inventories of Trade Suppliers, making sure to use the `slot` field to achieve the correct order (second page will need to be guessed).

Also I'd change all `<Enchanting Supplies>` subnames to `<Enchanter>`.

**Comment by mindphluxnet** [comment]
Here are the remaining items from page 2+ - any idea how to sort them?

Black Dye
Blue Dye
Bright Baubles
Coal
Crystal Vial
Dust of Decay
Enchanted Powder
Fine Thread
Gray Dye
Green Dye
Heavy Silken Thread
Heavy Stock
Hot Spices
Leaded Vial
Lethargy Root
Nightcrawlers
Purple Dye
Silken Thread
Soothing Spices
Strong Flux
Wooden Stock
Yellow Dye

**Comment by GrenderG** [comment]
The order that appears here seems to be the correct one according to the first page that we have: https://web.archive.org/web/20060306111845/http://wow.allakhazam.com/db/mob.html?wmob=1286

**Comment by mindphluxnet** [comment]
Great find, I've added them all now.

File: etc/databases/world/updates/updates.sql [hunk 1]
```diff
@@ -30418,6 +30418,85 @@ begin not atomic
...
+        update npc_vendor set slot = 30 where item = 8925 and entry in(2118, 5163, 4561, 1250, 1286, 5100, 3168, 843, 4897, 4877, 4229, 3614, 3499, 3187, 3482, 66, 4194);
+        update npc_vendor set slot = 31 where item = 4400 and entry in(2118, 5163, 4561, 1250, 1286, 5100, 3168, 843, 4897, 4877, 4229, 3614, 3499, 3187, 3482, 66, 4194);
+        update npc_vendor set slot = 32 where item = 6532 and entry in(2118, 5163, 4561, 1250, 1286, 5100, 3168, 843, 4897, 4877, 4229, 3614, 3499, 3187, 3482, 66, 4194);
+        update npc_vendor set slot = 33 where item = 3372 and entry in(2118, 5163, 4561, 1250, 1286, 5100, 3168, 843, 4897, 4877, 4229, 3614, 3499, 3187, 3482, 66, 4194);
+        update npc_vendor set slot = 34 where item = 4399 and entry in(2118, 5163, 4561, 1250, 1286, 5100, 3168, 843, 4897, 4877, 4229, 3614, 3499, 3187, 3482, 66, 4194);
+        update npc_vendor set slot = 35 where item = 6530 and entry in(2118, 5163, 4561, 1250, 1286, 5100, 3168, 843, 4897, 4877, 4229, 3614, 3499, 3187, 3482, 66, 4194);
+
+        -- Set the correct subname for Leo Sarn.
+        update creature_template set subname = 'Enchanter' where entry = 5758;
```
Comment (GrenderG) [inline etc/databases/world/updates/updates.sql:30495]: Please change subname for all `<Enchanting Supplies>` NPCs.

Comment (mindphluxnet) [reply etc/databases/world/updates/updates.sql:30495]: I did ... that was in another PR you already merged :) I'll remove the line if need be.

Comment (GrenderG) [reply etc/databases/world/updates/updates.sql:30495]: > I did ... that was in another PR you already merged :) I've removed that query.

In which PR? I can't find the query in `updates.sql`.

Comment (mindphluxnet) [reply etc/databases/world/updates/updates.sql:30495]: Hm, neither can I anymore. Seems to have vanished. I'll readd it.

Comment (mindphluxnet) [reply etc/databases/world/updates/updates.sql:30495]: Done

Comment (GrenderG) [reply etc/databases/world/updates/updates.sql:30495]: Thanks!

**Comment by GrenderG** [comment]
I'm thinking that probably all Trade Suppliers shared the same inventory back in the day. For example Tharynn Bouden's inventory currently doesn't match the original screenshot we have (In Vanilla there were some differences between inventories of Trade Suppliers).

Current:
image

Original:
image

If you take a look, you will notice that the original screenshots shows the same generic inventory for all Trade Suppliers (see screenshots at https://github.com/The-Alpha-Project/alpha-core/issues/920#issuecomment-1456399294).

So... maybe inventories need to be removed and just create a new one from scratch and use it for all of them. What do you think @mindphluxnet?

Pinging also @geo-tp and @GizzleBizzle here.

**Comment by mindphluxnet** [comment]
Pretty sure you're correct there. If others agree I can setup a new inventory.

**Comment by GrenderG** [comment]
> images_1488 images_2343 images_6571
> 
> Screenshots with the same vendor list. Second screenshot doesnt show the exact same order.

NPC from second screenshot is not spawned in 0.5.3 though.

**Comment by mindphluxnet** [comment]
I'd like to make a new PR with the unified tradeskill vendor inventory if possible.

**Pull request merged**